### PR TITLE
crc-libvirt: Use conf.d directory

### PR DESCRIPTION
**Pull request by cgwalters**
Use the `conf.d` directory rather than changing the main config
which will be fragile.

Tweak things to be more idempotent too.

### CODE REVIEW

**Comment by cgwalters** [comment]
(Totally untested)

**Comment by praveenkumar** [comment]
Thanks, I will test that tomorrow and merge it.

File: libvirt/crc_libvirt.sh [hunk 1]
```diff
@@ -56,11 +56,18 @@ EOF
...
-    if [ $? -ne 0 ]
-    then
-        sudo sed -i.bak '/\[main\]/a dns=dnsmasq' /etc/NetworkManager/NetworkManager.conf
-        echo server=/tt.testing/192.168.126.1 | sudo tee /etc/NetworkManager/dnsmasq.d/openshift.conf
+    dnsconf=/etc/NetworkManager/conf.d/crc-libvirt-dnsmasq.conf
+    local dnschanged=""
+    if ! [ -f "${dnsconf}" ]; then
+        echo 'dns=dnsmasq' | sudo tee "${dnsconf}"
+        dnschanged=1
+    fi
```
Comment (zeenix) [inline libvirt/crc_libvirt.sh]: I tried this and NetworkManager didn't start after that: `Feb 13 16:13:59 linux.fritz.box NetworkManager[29948]: Failed to read configuration: /etc/NetworkManager/conf.d/crc-libvirt-dnsmasq.conf: id=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=NetworkManager comm="systemd" exe="/usr/lib/systemd/systemd" hostnat=NetworkManager comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'`

Comment (cgwalters) [reply libvirt/crc_libvirt.sh]: Oops need `[main]` before that

**Comment by gbraad** [comment]
You might also wanna change this in the installer libvirt-howto.md

**Comment by zeenix** [comment]
> You might also wanna change this in the installer libvirt-howto.md

Done! @cgwalters would you update your patch according to those changes or shall i create another PR?

**Comment by gbraad** [comment]
Which changes are you suggesting?

**Comment by zeenix** [comment]
> Which changes are you suggesting?

At least the '[main]' is missing from Collin's patch. :)

**Comment by gbraad** [comment]
I think you can add this as a patch on top or rebase/amend. Colin is
usually quite busy...

**Comment by zeenix** [comment]
> I think you can add this as a patch on top or rebase/amend.

I can push to his branch?

**Comment by gbraad** [comment]
this happened, right?

**Comment by zeenix** [comment]
> this happened, right?

nope! it was forgotten.

**Comment by zeenix** [comment]
Now we can close this in favour of #49.

**Pull request closed without merging**